Sources/NostrSDK/Tag.swift

```diff
@@ -8,51 +8,82 @@
 import Foundation
 
 /// A constant that describes the type of a ``Tag``.
-public enum TagName: Codable, Equatable {
+public enum TagName: Codable, Equatable, CaseIterable {
+    
+    /// a custom emoji that defines the shortcode name and image URL of the image file
+    case emoji
     
     /// points to the id of an event this event is quoting, replying to or referring to somehow
     case event
     
+    /// a hashtag to categorize events for easy searching
+    case hashtag
+    
     /// points to a pubkey of someone that is referred to in the event
     case pubkey
     
+    case publishedAt
```

Comment:
As mentioned in your PR description, adding tags here for now should be fine. But I foresee a better solution with moving to kind-specific tags in the specific Nostr event class file, rather than having a universal catch-all.